pkgs/build-support/go/module.nix

```
@@ -84,6 +84,7 @@ let
     preBuild = args.preBuild or "";
     postBuild = args.modPostBuild or "";
     sourceRoot = args.sourceRoot or "";
+    ${if args?setSourceRoot then "setSourceRoot" else null} = args.setSourceRoot;
```

[`pkgs/stdenv/generic/setup.sh`](https://github.com/NixOS/nixpkgs/blob/3af282374194ad97fd9642e7245f6264631282a2/pkgs/stdenv/generic/setup.sh#L1138) behaves the same when `setSourceRoot` is `""`, `null`, or does not exist.

Considering that
- [Fixed-point arguments](https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes), a potential way to fix `overrideAttrs` for Go packages, requires an attribute to exist, and
- `""` is friendlier to `overrideAttrs` than `null` for strings containing build commands,

it would  be better make `setSourceRoot` default to `""` if not set.

```suggestion
    setSourceRoot = args.setSourceRoot or "";
```

---

Done.